packages/mui-material/src/Divider/Divider.js

```diff
@@ -54,100 +55,136 @@ const DividerRoot = styled('div', {
-    borderStyle: 'solid',
-    borderColor: (theme.vars || theme).palette.divider,
-    borderBottomWidth: 'thin',
-    ...(ownerState.absolute && {
-      position: 'absolute',
-      bottom: 0,
-      left: 0,
-      width: '100%',
-    }),
-    ...(ownerState.light && {
-      borderColor: theme.vars
-        ? `rgba(${theme.vars.palette.dividerChannel} / 0.08)`
-        : alpha(theme.palette.divider, 0.08),
-    }),
-    ...(ownerState.variant === 'inset' && {
-      marginLeft: 72,
-    }),
-    ...(ownerState.variant === 'middle' &&
-      ownerState.orientation === 'horizontal' && {
+})(({ theme }) => ({
+  margin: 0, // Reset browser default style.
+  flexShrink: 0,
+  borderWidth: 0,
+  borderStyle: 'solid',
+  borderColor: (theme.vars || theme).palette.divider,
+  borderBottomWidth: 'thin',
+  variants: [
+    {
+      props: {
+        absolute: true,
+      },
+      style: {
+        position: 'absolute',
+        bottom: 0,
+        left: 0,
+        width: '100%',
+      },
+    },
+    {
+      props: {
+        light: true,
+      },
+      style: {
+        borderColor: theme.vars
+          ? `rgba(${theme.vars.palette.dividerChannel} / 0.08)`
+          : alpha(theme.palette.divider, 0.08),
+      },
+    },
+    {
+      props: {
+        variant: 'inset',
+      },
+      style: {
+        marginLeft: 72,
+      },
+    },
+    {
+      props: ({ ownerState }) =>
+        ownerState.variant === 'middle' && ownerState.orientation === 'horizontal',
+      style: {
         marginLeft: theme.spacing(2),
         marginRight: theme.spacing(2),
-      }),
-    ...(ownerState.variant === 'middle' &&
-      ownerState.orientation === 'vertical' && {
+      },
+    },
+    {
+      props: ({ ownerState }) =>
+        ownerState.variant === 'middle' && ownerState.orientation === 'vertical',
+      style: {
         marginTop: theme.spacing(1),
         marginBottom: theme.spacing(1),
-      }),
-    ...(ownerState.orientation === 'vertical' && {
-      height: '100%',
-      borderBottomWidth: 0,
-      borderRightWidth: 'thin',
-    }),
-    ...(ownerState.flexItem && {
-      alignSelf: 'stretch',
-      height: 'auto',
-    }),
-  }),
-  ({ ownerState }) => ({
-    ...(ownerState.children && {
-      display: 'flex',
-      whiteSpace: 'nowrap',
-      textAlign: 'center',
-      border: 0,
-      '&::before, &::after': {
-        content: '""',
-        alignSelf: 'center',
       },
-    }),
-  }),
-  ({ theme, ownerState }) => ({
-    ...(ownerState.children &&
-      ownerState.orientation !== 'vertical' && {
+    },
+    {
+      props: {
+        orientation: 'vertical',
+      },
+      style: {
+        height: '100%',
+        borderBottomWidth: 0,
+        borderRightWidth: 'thin',
+      },
+    },
+    {
+      props: {
+        flexItem: true,
+      },
+      style: {
+        alignSelf: 'stretch',
+        height: 'auto',
+      },
+    },
+    {
+      props: ({ ownerState }) => ownerState.children,
```

Comment:
```suggestion
      props: ({ ownerState }) => !!ownerState.children,
```
should explicitly return `boolean`.

Reply:
Note: this could be replaced with pseudo `:not(:empty)`